common/concurrency/limits/src/main/java/io/helidon/common/concurrency/limits/AimdLimitImpl.java

```diff
@@ -100,22 +102,42 @@ int currentLimit() {
     }
 
     Optional<LimitAlgorithm.Token> tryAcquire(boolean wait) {
+        return tryAcquire(wait, outcome -> {});
+    }
+
+    Optional<LimitAlgorithm.Token> tryAcquire(boolean wait, Consumer<LimitOutcome> outcomeConsumer) {
+        Objects.requireNonNull(outcomeConsumer, "limit algorithm outcome consumer cannot be null");
         Optional<LimitAlgorithm.Token> token = handler.tryAcquire(false);
         if (token.isPresent()) {
+            LimitOutcomeImpl.Accepted outcome = LimitOutcomeImpl.createAccepted(originName, AimdLimit.TYPE);
+            if (token.get() instanceof OutcomeAwareToken outcomeAwareToken) {
+                outcomeAwareToken.outcome(outcome);
+            }
+            outcomeConsumer.accept(outcome);
             return token;
```

Comment:
All this work above is wasted if the consumer is empty, right? Could this be avoided?

Reply:
We could refactor this code a bit it so a private method accepts a null for the consumer and not use it if it is null.

Reply:
The new approach goes to extensive lengths to avoid work if it does not apply. Among them, internally the code uses `null` to indicate the lack of a consumer, avoiding  whatever overhead from invoking an empty consumer might not be optimized away.